benchmarks/protoboeuf/benchmark_pb.rb

```diff
@@ -0,0 +1,4888 @@
+# frozen_string_literal: true
+module ProtoBoeuf
+class Position
+  def self.decode(buff)
+    buff = buff.dup
+    buff.force_encoding("UTF-8")
+    allocate.decode_from(buff, 0, buff.bytesize)
+  end
+
+  # required field readers
+  attr_accessor :x, :y, :z
+
+  # enum readers
+
+  # enum writers
+
+  def initialize(x: 0, y: 0, z: 0)
+    @x = x
+    @y = y
+    @z = z
+  end
+
+  def decode_from(buff, index, len)
+    
+      @x = 0
+      @y = 0
+      @z = 0
+
+            tag = buff.getbyte(index)
+        index += 1
+
+
+    while true
+      if tag == 0x8
+                ## PULL_UINT64
+        @x =       if (byte0 = buff.getbyte(index)) < 0x80
+        index += 1
+        byte0
+      else
+        if (byte1 = buff.getbyte(index + 1)) < 0x80
```

Comment:
BTW this logic looks like protobuf varint deserialization, is it faster to have this verbose code compared to this simple loop I wrote in Prism?
https://github.com/ruby/prism/blob/20e768d8d75b78147e9588e0b842dbd847b20cfe/templates/lib/prism/serialize.rb.erb#L185-L199